### ISSUE TITLE
Create easy-to-discover-and-use upgrade refactoring UI for the move from annotations to keyword fields.

### ISSUE THREAD

**Issue by jurgenvinju** [usethesource/rascal-language-servers#392]
The `update` functionality in lang::rascal::upgrade::* on the current branch (remove-tree-loc-annotations) that refactors uses of annotation syntax to keyword parameter syntax is a candidate for inclusion in the Rascal LSP server:
* as a lense
* as a refactoring?

It has to run on the entire project to make sense as the refactoring is "dumb"; it does not follow name resolution links it just replaces all syntactical constructs in a consistent manner.

_Originally posted by @jurgenvinju in https://github.com/usethesource/rascal/issues/1974#issuecomment-2162825547_

**Comment by DavyLandman** [comment]
Agreed, but I would be really interested in if we could do something where we don't bother users that are either not using annotations, or have already done the migration for a specific project.

**Comment by jurgenvinju** [comment]
We could add a lense at the top of every module that still uses annotations, and otherwise not at all. And optionally also add the same lense command near every use of annotation syntax? 

The hover help should be insightful enough to explain that the lense will rewrite the entire project. @DavyLandman Do we have hover help on lenses already?

I've already integrated a file-by-file based progress bar (it could take a few seconds if the project is big).

**Comment by urbanfly** [comment]
IMO, refactoring / upgrade suggestions make more sense as CodeActions than CodeLenses

**Comment by DavyLandman** [comment]
Agreed, a warning/error with a code action attached to it. But we have to figure out if we can identify all places with a specific warning.

**Comment by DavyLandman** [comment]
A quickfix has been added in #1099 

The warning comes from the typechecker, so this can take a while to pop up, we could consider some kind of extra warning detector that runs much quicker, such that users get this warning before pressing save.
